### PR TITLE
initial commit for assignment 1

### PR DESCRIPTION
**Pull request by meta-aman-gautam**

### CODE REVIEW

File: Item.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+public class Item {
```
Comment (meta-prateek-jain) [inline Item.java]: Missing Comments

File: Item.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+	public String getProduct_itemname() {
+		return product_itemname;
+	}
+
+	public double getProduct_price() {
+		return product_price;
+	}
+
+
```
Comment (meta-prateek-jain) [inline Item.java]: Extra Empty Spaces

File: ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	 * @param {string}item -item is the object of class Item
+	 * @param {integer}quantity
+	 */
+	public static void add(Item item, int quantity) {
+		if (cart.contains(item)) {
+			System.out.println("This Item already exists");
+
+		} else {
+
+			if (quantity > 0) {
```
Comment (meta-prateek-jain) [inline ShoppingCart.java]: Can define this as `else if(quantity > 0)`

File: ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+	
+	/**
+	 * showCart method is used to DISPLAY items of the cart.
+	 */
+	public static void showCart() {
+		
+		System.out.println("\nItem Name\t" + "Item Qty\t"+ "Item price(/piece)");
+		System.out.println("==========\t" + "========\t" + "==================");
+		for (Item value : cart) {
```
Comment (meta-prateek-jain) [inline ShoppingCart.java]: A variable name should be meaningful. E.g. `value` can be  `cartItem`

File: ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		System.out.println("=========\t" + "========\t" + "=============\t"
+				+ "=====");
+		for (Item value : cart) {
+			System.out.println(value.product_itemname + "   \t   "
+							+ value.product_quantity  + "   \t   "
+							+ value.product_price     + "   \t   "
+							+ (value.product_price * value.product_quantity));
+
+			/*
+			 * caluculating total price of the all the products .
```
Comment (meta-prateek-jain) [inline ShoppingCart.java]: Here single line comment could be used

File: ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+		System.out.println("=========\t" + "========\t" + "=============\t"
+				+ "=====");
+
+		System.out.println("\nTOTAL = " + total + "\n");
+	}
+
+	
+	/**
+	 * Main method
```
Comment (meta-prateek-jain) [inline ShoppingCart.java]: Comment message can be improved. Something like `Initialize the program`

File: ShoppingCart.java [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+				default:
+					System.out.println(" XXX Invalid Choice ...pls try again XXX ");
+
+				}
+
+			}
+
+			else if (menu == 2) {
+				System.out
+						.print("Choose from Available Option to update/remove items");
```
Comment (meta-prateek-jain) [inline ShoppingCart.java]: Code formatting

File: ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+	/*
+	 * Object of class Item(pencil)
+	 */
+	static Item pencil = new Item("pencil", 5.0);
+	static Item eraser = new Item("eraser", 3.5);
+	static Item sharpner = new Item("sharpner", 6.9);
+	static Item scale = new Item("scale", 25.0);
+	static Item notebook = new Item("notebook", 55.5);
+
```
[the remark this answers is not in the document]
Comment (meta-aman-gautam) [reply ShoppingCart.java]: okays sir .

File: ShoppingCart.java (hunk 7 above)
Comment (meta-aman-gautam) [reply ShoppingCart.java]: okay sir , will change immediately

File: ShoppingCart.java (hunk 6 above)
Comment (meta-aman-gautam) [reply ShoppingCart.java]: okay sir

File: ShoppingCart.java (hunk 5 above)
Comment (meta-aman-gautam) [reply ShoppingCart.java]: okay sir

File: Item.java (hunk 1 above)
Comment (meta-aman-gautam) [reply Item.java]: comments added

File: Item.java (hunk 2 above)
Comment (meta-aman-gautam) [reply Item.java]: Empty spaces removed

File: ShoppingCart.java (hunk 3 above)
Comment (meta-aman-gautam) [reply ShoppingCart.java]: optimization done

File: ShoppingCart.java (hunk 4 above)
Comment (meta-aman-gautam) [reply ShoppingCart.java]: variable name changed

**Pull request merged**